gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,877 @@
+        "\n",
+        "*   def check_info(resume: str) -> Dict[str, str]: Uses Gemini to extract name, address, mobile number, and email from a resume, returning 'Not Found' if information is missing.\n",
+        "    \n",
+        "*   def get_info(resume: str) -> str: Similar to check_info but returns the extracted information as a formatted string.\n",
+        "\n",
+        "*   def iter_block_items(parent: Table) -> str:\n",
+        "    Iterates through a docx file object, yielding paragraphs and tables.\n",
+        "\n",
+        "*   def convert_docx_to_text(file_path: str) -> str:\n",
+        "    Parses a docx file and converts it to text, handling tables and paragraphs.\n",
+        "\n",
+        "*   def parse_resume(res_path: str) -> str:\n",
+        "    Parses a resume from various file types (docx, pdf, txt) with fallbacks to handle missing personal information.\n",
+        "\n",
+        "*   def get_response(parsed_res: str, parsed_jd: str) -> str:\n",
+        "    Generates a resume using Gemini, incorporating extracted information, a professional summary, and arranged skills/coursework.\n",
+        "\n",
+        "*   def structure_response(resume_generated: str) -> str:\n",
+        "    Structures the generated resume into a specific format for readability."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "c8dc8fe8-890c-4471-915a-396787e84a1d"
+      },
+      "outputs": [],
+      "source": [
+        "def check_info(resume: str) -> dict[str, str]:\n",
+        "\n",
+        "    vertexai.init(\n",
+        "        project=\"aniket-tdcx\", location=\"us-central1\"\n",
+        "    )  # Initialize the AI Platform client with project ID\n",
+        "\n",
+        "    \"\"\"\n",
+        "    Check if the name, address, mobile number and email address is present in the resume\n",
+        "\n",
+        "    Args:\n",
+        "        resume: The parsed resume as str\n",
+        "\n",
+        "    Returns:\n",
+        "        Details requested in the prompt as dict\n",
+        "    \"\"\"\n",
+        "    prompt = f\"\"\"\n",
+        "    Extract the following candidate information from the given resume\n",
+        "    1. Name\n",
+        "    2. Address\n",
+        "    3. Mobile Number\n",
+        "    4. Email address\n",
+        "    Text : {resume}\n",
+        "\n",
+        "    INSTRUCTIONS:\n",
+        "    - In case the information is not present, return 'Not Found'\n",
+        "    - The final output should be in valid JSON format\n",
+        "    \"\"\"\n",
+        "    resp = model.generate_content(prompt).text\n",
+        "    return literal_eval(resp.split(\"```json\")[-1].split(\"```\")[0].replace(\"\\n\", \"\"))"
```

Comment:
Same as before, Controlled Generation instead of raw markdown parsing.

Reply:
Replaced with GenerationConfig class and added controlled generation across the code